### PR TITLE
Mark edge cells for culling for non-periodic boundaries 

### PR DESCRIPTION
**Pull request by amrapallig**
Two logical arguments are added `-npx` and `-npy` to enable nonperiodic boundary condition in `x` (`--nonperiodic_x`) and `y` (`--nonperiodic_y`) directions, respectively, as required. If `nonperiodic_x: nx=nx+2`, if `nonperiodic_y: ny=ny+2`. Extra cells are added that are culled when non-periodic boundary conditions are invoked such that the number of cells is the same as input values.

### CODE REVIEW

**Comment by amrapallig** [comment]
This was tested as follows:
Planar_hex.py ran using python 2.7 as it showed error with load_latest_e3sm_unified.sh
`case1: python planar_hex.py --nx 6 --ny 8 --dc 10000  -o base_mesh.nc`
`periodic boundaries with nx=6 ny=8`
`The figure shows all the cells`
pic1


`case2:  python planar_hex.py --nx 6 --ny 8 --dc 10000  -npx  -o base_mesh.nc`
`non-periodic boundary in x  with nx=6 ny=8`
`The figure shows cullCell, red indicates the extra cells added to be culled for non-periodic boundary conditions`
pic2


`case3:  python planar_hex.py --nx 6 --ny 8 --dc 10000  -npy -o base_mesh.nc`
`non-periodic boundary in y  with nx=6 ny=8`
`The figure shows cullCell, red indicates the extra cells added to be culled for non-periodic boundary conditions`
pic3

 
`case4:  python planar_hex.py --nx 6 --ny 8 --dc 10000  -npx -npy -o base_mesh.nc`
`non-periodic boundaries in both x and y  with nx=6 ny=8`
`The figure shows cullCell, red indicates the extra cells added to be culled for non-periodic boundary conditions`
pic4

File: mesh_tools/planar_hex/planar_hex.py [hunk 1]
```diff
@@ -130,6 +144,34 @@ def initial_setup(nx, ny, dc):
...
+
+def mark_cull_cell_nonperiodic_x(mesh):
+
+    cullCell = mesh.cullCell
+    cellIdx = mesh.cellIdx
+    cellRow = mesh.cellRow
+    cellCol = mesh.cellCol
+    nCells = mesh.sizes['nCells']
+    nx = mesh.sizes['nx']
+    ny = mesh.sizes['ny']
```
Comment (xylar) [inline mesh_tools/planar_hex/planar_hex.py]: `ny` is not used, so please remove this line

File: mesh_tools/planar_hex/planar_hex.py [hunk 2]
```diff
@@ -130,6 +144,34 @@ def initial_setup(nx, ny, dc):
...
+    #print(nx,ny,nCells)
```
Comment (xylar) [inline mesh_tools/planar_hex/planar_hex.py]: please remove this line.

File: mesh_tools/planar_hex/planar_hex.py [hunk 3]
```diff
@@ -130,6 +144,34 @@ def initial_setup(nx, ny, dc):
...
+def mark_cull_cell_nonperiodic_y(mesh):
+
+    cullCell = mesh.cullCell
+
+    cellIdx = mesh.cellIdx
+    cellRow = mesh.cellRow
+    cellCol = mesh.cellCol
+    nCells = mesh.sizes['nCells']
+    nx = mesh.sizes['nx']
+    ny = mesh.sizes['ny']
```
Comment (xylar) [inline mesh_tools/planar_hex/planar_hex.py]: ny is not used so please remove this line.

File: mesh_tools/planar_hex/planar_hex.py [hunk 4]
```diff
@@ -404,18 +446,24 @@ def main():
...
+			help='non-periodic in  y direction')
     parser.add_argument('-o', '--outFileName', dest='outFileName', type=str,
                         required=False, default='grid.nc',
                         help='The name of the output file')
-
+       	             	
     # parser.add_argument('--periodicX', dest='periodicX', action='store_true',
     #                     help='Make the mesh periodic in x')
     # parser.add_argument('--periodicY', dest='periodicY', action='store_true',
     #                     help='Make the mesh periodic in y')
```
Comment (xylar) [inline mesh_tools/planar_hex/planar_hex.py]: Please remove these commented-out bits of code, since your work replaces them.

**Comment by xylar** [comment]
Yes, I noticed that and don't see a problem. One could just translate the mesh if needed.

**Comment by amrapallig** [comment]
Thank you @xylar

**Comment by xylar** [comment]
@mark-petersen, why don't you merge this when you're ready.  It's a modification to code that's not ocean specific but I think other developers will be happy with this change and the tool is new enough that I doubt it's part of anyone else's workflow yet.

**Comment by sbrus89** [comment]
I second @xylar's comment. I just used it yesterday and it worked great. Thanks @amrapallig!

**Comment by pwolfram** [comment]
I third-- no problem with merging and that seems the right approach.

**Comment by xylar** [comment]
Thanks again, @amrapallig.  You can now delete your remote branch.

**Comment by amrapallig** [comment]
Thanks all. I appreciate everyone's input.

**Pull request merged**